### PR TITLE
feat(router): emit Stripe-Account on UCS repeat-payment flow (#16727)

### LINKED ISSUE #13127: [shadow-validation] stripe/repeat_payment: UCS↔HS request diff
Shadow-validation flagged a connector-request diff for **stripe / repeat_payment**.

`router.typeDiff:connector_http_status_code`

Fixed by HS PR #12806.

### PR DESCRIPTION
**Pull request by 10X-GRACE**
Closes #13127

### CODE REVIEW

File: crates/router/src/core/unified_connector_service/transformers.rs [hunk 1]
```diff
@@ -2161,6 +2161,9 @@ impl
                 phone_country_code: None,
                 customer_document_details: to_grpc_customer_document_details(router_data),
             }),
```
Comment (XyneSpaces) [inline crates/router/src/core/unified_connector_service/transformers.rs:2163]: :warning: Verify `build_stripe_split_payment` handles `None` input gracefully. This path is exercised for all non-Stripe-Connect MIT payments; a panic or unexpected error here would break recurring payments.

Confirm the helper returns `Option<StripeSplitPaymentRequest>` and add a test for the `split_payments: None` case.

**Review by XyneSpaces — COMMENTED** [review]
## Review Summary

**Classification:** Core Flow Framework + Dependency Update (UCS Stripe Connect repeat-payment)
**Risk Level:** Medium-High (dependency rev-pin change affects payment flow)

---

### :warning: Warning — Dependency Pin Points to Branch, Not Tagged Release

**Location:** `Cargo.lock`, `crates/*/Cargo.toml`

All 4 `connector-service` dependencies are being bumped from a stable rev (`ce095385623e4e5ed87e00eec06c5667a0fdd189`) to a branch rev (`0c3a17b45d4290971a9aa8f76bf8a74f633b00bd`).

Per the PR description, this currently points at the proto-rev branch `ucs-diff/stripe/16727-repeat-proto`. Shipping a branch pin to production creates an unstable dependency that may change unexpectedly if the branch is rebased or force-pushed.

**Fix:** Flip the rev-pin to a CalVer tag (e.g., `v2025.06.xx`) once the prism PR (juspay/hyperswitch-prism#1551) merges and the nightly tag job runs. Do not merge with a branch-based rev.



---

### :bulb: Suggestion — Verify build_stripe_split_payment Handles None Gracefully

**Location:** `crates/router/src/core/unified_connector_service/transformers.rs:2161`

The new field uses:
```rust
stripe_split_payment: build_stripe_split_payment(
    router_data.request.split_payments.as_ref(),
),
```

Ensure `build_stripe_split_payment` handles `None` appropriately for non-Stripe-Connect payments without requiring a `Some(...)` wrapper that might fail on `split_payments: None`. The helper should return `Option<StripeSplitPaymentRequest>` cleanly.

**Fix:** Confirm `build_stripe_split_payment` signature returns `Option<_>` and handles `None` input gracefully. Add a unit test for the `split_payments: None` path.



---

### :mag: Nitpick — Test Coverage for RecurringPaymentServiceChargeRequest

**Location:** Test suite

The diff does not include any test updates for the `RecurringPaymentServiceChargeRequest` construction. Given this is a payment-critical path (MIT recurring charges), ensure there is test coverage verifying:
1. `stripe_split_payment` is correctly populated when `split_payments` is present
2. The field is absent/None when `split_payments` is absent

**Fix:** Add or reference the test file that exercises this transformer. If tests exist in a separate PR, link them in the PR description.



---

**Verdict:** :arrows_counterclockwise: **Request Changes** — Address the dependency branch pin before merge.

**Review by XyneSpaces — COMMENTED** [review]
## Review Summary

**Verdict:** :white_check_mark: Approve

**Classification:** Core flow framework change — UCS repeat-payment Stripe Connect support  
**Risk:** Low

The PR correctly threads `stripe_split_payment` through the UCS repeat-payment flow to enable Stripe Connect direct charges. Changes are minimal and follow established patterns.

---

### Findings

No blocking issues identified. The change:
- Properly uses `build_stripe_split_payment()` helper to transform split payments data
- Correctly updates all 4 `connector-service` dependency pins to the same revision
- Places the field assignment alongside other request data population in the transformer

### Verification Completed

- [x] Rev-pin is consistent across Cargo.toml files (external_services, hyperswitch_interfaces, router)
- [x] Cargo.lock updated with matching revisions
- [x] Proto field addition (field 34) aligns with companion PR in hyperswitch-prism#1551
- [x] Shadow validation evidence provided in PR description confirms header parity
